docs/docsite/rst/playbook_guide/playbooks_reuse_roles.rst

```diff
@@ -281,63 +281,100 @@ role ``meta/argument_specs.yml`` file. All fields are lower-case.
     * This should be ``main`` in the case of an unspecified entry point.
     * This will be the base name of the tasks file to execute, with no ``.yml`` or ``.yaml`` file extension.
 
-    :short_description:
+.. list-table:: Role Argument Specification Fields
+  :width: 90%
+  :widths: auto
+  :header-rows: 1
+
+  * - Name
+    - Description
+    - Type
+    - Required
+    - Default
+  * - short_description
+    - A short, one-line description of the entry point.
+    - str
+    - Yes
+    -  
+  * - description
+    - A longer description that may contain multiple lines.
+    - list\n elements: str
+    - No (see notes)
+    - null
+  * - author
+    - Name of the entry point authors.
+    - list\n elements: str
+    - No
+    - []
+  * - options
+    - Options are often called "parameters" or "arguments".
+    - dict (see table below)
+    - No
+    - {}
+
+
+Options themselves have a defined structure
+
+.. list-table:: Role Argument Specification options
+  :width: 90%
+  :widths: auto
+  :header-rows: 1
+
+  * - <Name> (the option itself)
+    - Description
+    - Type
+    - Required
+    - Default
+  * - name
+    - The name of the option/argument.
+    - str
+    - Yes
+    -   
+  * - description
+    - Detailed explanation of what this option does. It should be written in full sentences.
+    - list\n elements: str
+    - No
+    - null
+  * - type
+    - The data type of the option. See :ref:`Argument spec <argument_spec>` for allowed values for ``type``.
+    - No
+    - str
+    - 
+  * - required
+    - Denote if the option is required or not
+    - bool
+    - No
+    - false
+  * - default
+    - If an option is not required, the value is set to this if not otherwise provided
+    - any (either null or should match Type above)
+    - No
+    - null
+  * - choices
+    - List of option values.
+    - list
+    - No
+    - []
+  * - elements
+    - Specifies the data type for list elements when type is ``list``.
+    - str
+    - No
+    - null
+  * - options
+    - If this option takes a dict or list of dicts, you can define the structure here.
+    - dict (recursively use this same table)
+    - No
+    - null
```

Comment:
```suggestion
    - null
  * - version_added
    - For roles in collections, the version of the collection this option was added in.
      Should only be specified if this is different from the ``version_added`` value of the entrypoint itself.
    - str
    - No
    -
```

Reply:
need to x2 check the code, but there does not seem to be any handling in core for this

Reply:
It does work, `ansible-doc -t role felixfontein.acme.acme_certificate` shows `added in: version 0.6.0` for `acme_certificate_dns_substitution`.

Reply:
that is expected as `ansible-doc` is permissive and will not filter out what it does not know, i was checking the spec validator, but it does the same, it ignores unknown